content/swift/concepts/functions/functions.md

```diff
@@ -0,0 +1,140 @@
+Subjects: 
+  - 'iOS'
+  - 'Code Foundations'
+Tags:
+  - 'iOS'
+  - 'Algorithms'
+  - 'Functions'
+CatalogContent:
+  - 'learn-Swift-course'
+  - 'path/iOS-path'
+---
+
+**Functions** are self-contained chunks of code that perform a specific task. You give a function a name that identifies what it does, and this name is used to “call” the function to perform its task when needed.
+
+## Defining and Calling Functions
+Any funcation have three important parameters used to define it, first the funcation name, this is used to call the function, second is the number of parameters a function takes, parameters refers to the input the function take when calling it and lastly the return type of the function, this refer to the output data type of the function. A small tip when defining the name of the function, the name should describe it's functionality. 
+This below figure describe the syntax of funcations in swift:
+
+![This diagram explains the syntax of the functions in Swift ](https://raw.githubusercontent.com/Codecademy/docs/main/media/funcationsDiagram.png)
+
+```Swift
+func greet(person: String) -> String {
+    let greeting = "Hello, " + person + "!"
+    return greeting
+}
+```
+For example, this function called greet and takes a `String` parameter called person, and returns a `String` equale to Hello plus `person` value. Call the greet(person:) function by passing it a `String` value after the person argument label as shown below:
+
+```Swift
+print(greet(person: "Anna"))
+// Prints "Hello, Anna!"
+```
+
+## Multiple Parameters
+Functions can take more than one input parameter, the parameters must be separated by commas and assigned during the function call.
+For example, this function takes two numbers as an input, and returns the division of the two numbers:
+
+```Swift
+func division(numerator: Double, denominator: Double) -> Double {
+  return numerator / denominator
+} 
+
+let decimal = division(numerator: 1.0, denominator: 2.0) 
+print(decimal)
+ // Prints:  0.5 
+```
+
+## Returning Multiple Values
+You can use a tuple type as the return type for a function to return multiple values as part of one compound return value. Each value within its parentheses needs to be labeled and assigned a type in the function definition.
+For example, this function takes array of `Int` as an input, and return the smallest and largest number in the array.
+```Swift
+func minMax(array: [Int]) -> (min: Int, max: Int) {
+    var currentMin = array[0]
+    var currentMax = array[0]
+    for value in array[1..<array.count] {
+        if value < currentMin {
+            currentMin = value
+        } else if value > currentMax {
+            currentMax = value
+        }
+    }
+    return (currentMin, currentMax)
+}
+```
+Because the tuple’s member values are named as part of the function’s return type, they can be accessed with dot syntax to retrieve the minimum and maximum found values, as done below:
```

Comment:
```suggestion

Because the tuple’s member values are named as part of the function’s return type, they can be accessed with dot syntax to retrieve the minimum and maximum found values, as done below:

```